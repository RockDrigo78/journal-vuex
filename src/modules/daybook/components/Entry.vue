<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: 10 } })"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">December</span>
      <span class="mx-1 fs-5">31</span>
      <span class="mx-2 fw-light">2021</span>
    </div>
    <div class="entry-description">
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Nulla
      voluptatibus dolore exercitationem veniam quaerat, enim id nihil
      doloremque temporibus! Esse iste voluptas at reiciendis officiis iure
      velit sed quaerat distinctio.
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang='scss' scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all eaase-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>