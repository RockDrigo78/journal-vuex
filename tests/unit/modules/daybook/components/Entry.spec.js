import { shallowMount } from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry.vue';
import { journalState } from '../../../mock-data/test-journal-state';

describe('Entry component tests', () => {
    const mockRouter = {
        push: jest.fn(),
    };

    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0],
        },
        global: {
            mocks: {
                $router: mockRouter,
            },
        },
    });
    test('should match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('should redirect on click', () => {
        const entryContainer = wrapper.find('.entry-container');

        entryContainer.trigger('click');

        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id,
            },
        });
    });

    test('should get computed properties', () => {
        expect(wrapper.vm.day).toBe(23);
        expect(wrapper.vm.month).toBe('July');
        expect(wrapper.vm.year).toBe(2021);
        expect(wrapper.vm.weekday).toBe('Friday');
    });
});
