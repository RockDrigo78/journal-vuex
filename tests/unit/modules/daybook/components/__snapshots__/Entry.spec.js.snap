// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component tests should match snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-1 fs-5">July</span><span class="mx-2 fw-light">2021,</span><span class="mx-2 fw-light">Friday</span></div>
  <div class="entry-description">hello from mock data 1</div>
</div>
`;
